crates/core/tedge_agent/src/file_transfer_server/error.rs

```diff
@@ -24,8 +23,67 @@ pub enum FileTransferError {
     BindingAddressInUse { address: std::net::SocketAddr },
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum FileTransferRequestError {
+    #[error(transparent)]
+    FromIo(#[from] std::io::Error),
+
+    #[error("Cannot delete: {path:?} is a directory, not a file")]
+    CannotDeleteDirectory { path: RequestPath },
+
+    #[error("Cannot upload: {path:?} is a directory, not a file")]
+    CannotUploadDirectory { path: RequestPath },
+
+    #[error("Request to delete {path:?} failed: {source}")]
```

Comment:
note: We need to be careful with including source error message in parent error message.

I think the recommendation is to not include source error messages like this because proper error reporting/diagnostics will themselves call `.source()` to print messages for the entire error chain. However, for code that is not aware of this, and doesn't depend on any error reporting crates, it will most likely just grab outer error message. It depends where the error is used. For the FTS, I think the errors appear only in HTTP responses and logs, but 1) I'm not sure how they are exactly reported by axum 2) it's all internal so it doesn't matter that much.